### PR TITLE
allow multiple spec files to be provided to XcodeGen

### PR DESCRIPTION
**Pull request by skofgar**
**TL;DR: Generate 10 projects in less than 1 second instead of 8 by allowing XcodeGen to accept multiple project files instead of generating them one by one.**

When generating multiple projects it can take a long time. For example, when generating 10 projects by invoking XcodeGen 10 times sequentially via bash, the script takes about `8 seconds`, even with caching turned on. 

## Why is this a problem?

When switching branches and checking out new branches of a repo that contains multiple XcodeGen project files, this causes substantial delays and slows down engineers.

## Investigation

Looking at the profiler, it looks like the slowest part is just starting up XcodeGen. The "initialization" period can take up to `0.5 seconds`.

## Proposed Solution

Allowing XcodeGen to accept multiple spec files, can significantly speed up the generation. In our tests generation of the same 8 projects was less than `3 seconds` without caching and with caching less than `1 second` (sometimes just 0.2 seconds).

## Approach

 * allow specs to be provided as a comma separated list of specs instead of just one

```
xcodegen --spec path/to/first/project.yml,path/to/another/project.yml --use-cache
```

### CODE REVIEW

**Comment by skofgar** [comment]
Hi @yonaskolb may I request a review for this PR?

**Review by yonaskolb — APPROVED** [review]
Very cool, thanks @skofgar!
